File: rscm/trunk/ext/java/src/test/org/rubyforge/rscm/starteam/StarTeamTest.java
package org.rubyforge.rscm.starteam;

import junit.framework.TestCase;
import org.rubyforge.rscm.RSCM;
import org.rubyforge.rscm.YamlDumpable;

import java.io.IOException;
import java.io.PrintWriter;
import java.io.StringWriter;

/**
 * @author Aslak Helles&oslash;y
 */
public class StarTeamTest extends TestCase {
    private String userName = System.getProperty("starteam_user");
    private String password = System.getProperty("starteam_password");
    private String serverName = "192.168.254.21";
    private String serverPort = "49201";
    private String projectName = "NGST Application";
    private String viewName = "NGST Application";
    private String folderName = "data";
    private RSCM starTeam;

    protected void setUp() throws Exception {
        starTeam = new StarTeam(userName, password, serverName, serverPort, projectName, viewName, folderName);
//        starTeam = new TestScm(null, null);
    }

    public void testShouldConvertChangesToYaml() throws IOException {
        YamlDumpable changeSets = starTeam.getChangeSets("4 Jan 2005 04:02:00 -0000", "4 Jan 2005 04:26:00 -0000");
        final PrintWriter out = new PrintWriter(System.out);
        changeSets.dumpYaml(out);
        out.flush();
    }

    public void testShouldCheckout() throws IOException {
        YamlDumpable files = starTeam.checkout("target/starteam/checkout", null);
        String expected = "--- \n- eenie/meenie/minee/mo\n- catch/a/redneck/by\n- the/toe\n";
        StringWriter yaml = new StringWriter();
        files.dumpYaml(yaml);
        System.out.println(yaml);
        assertEquals(expected, yaml.toString());
    }
}

File: trunk/rscm/test/rscm/starteam/starteam_test.rb
require 'test/unit'
require 'fileutils'
require 'rscm/generic_scm_tests'
require 'rscm/starteam/starteam'

module RSCM
  class StarTeamTest < Test::Unit::TestCase
#    include GenericSCMTests

    def create_scm(repository_root_dir, path)
      StarTeam.new(ENV["STARTEAM_USER"], ENV["STARTEAM_PASS"], "192.168.254.21", 49201, "NGST Application", "NGST Application", "java")
    end

    def test_changesets
      from = Time.new - 2 * 3600 * 24
      to = Time.new - 1 * 3600 * 24
      puts "Getting changesets for #{from} - #{to}"
    
      changesets = create_scm(nil, nil).changesets(nil, from, to)
      assert_equal(1, changesets.length)
      assert_equal(Time.utc(2004, 11, 30, 04, 52, 24), changesets[0][0].time)
      assert_equal(Time.utc(2004, 11, 30, 04, 53, 23), changesets[0][1].time)
      assert_equal(Time.utc(2004, 11, 30, 04, 53, 23), changesets[0].time)
      assert_equal("rinkrank", changesets[0].developer)
      assert_equal("En to\ntre buksa \nned\n", changesets[0].message)
    end

    def test_checkout
      files = create_scm(nil, nil).checkout("target/starteam/checkout")
      assert_equal(3, files.length)
      assert_equal("eenie/meenie/minee/mo", files[0])
      assert_equal("catch/a/redneck/by", files[1])
      assert_equal("the/toe", files[2])
    end
  end
end
